clients/rospy/src/rospy/topics.py

```diff
@@ -1136,6 +1166,15 @@ def close_all(self):
                 t.close()
             self.pubs.clear()
             self.subs.clear()        
+
+            
+    def check_all(self):
+        """
+        Check all registered publication and subscriptions.
+        """
+        with self.lock:
+            for t in chain(iter(self.pubs.values()), iter(self.subs.values())):
+                t.check()
```

Comment:
The `check()` might alter `self.pubs` / `self.subs` those this might result in a `RuntimeError: dictionary changed size during iteration`. See #1894 for the proposed fix.